tests/test_logger.py

```diff
@@ -0,0 +1,170 @@
+    logging.root.setLevel(root_level)
+
+    if logging_plugin:
+        pytestconfig.pluginmanager.register(logging_plugin, "logging-plugin")
+
+
+@pytest.fixture(autouse=True)
+def reset_logging_module():
+    """Reset the logging specific configurations such as handlers or levels for
+    the module specific loggers."""
+    for module in modules_contain_logging:
+        logger = logging.getLogger(module.__name__)
+        logger.handlers = []
+        logger.setLevel(logging.NOTSET)
+
+
+@pytest.mark.parametrize("module", modules_contain_logging)
+class TestLogger:
+    """Tests for the functions that are used to create a logger"""
+
+    def test_logging_handler_defined(self, module):
+        """Tests that the logging_handler_defined function works correctly in
+        the following cases:
+
+        1. When a custom logger was just created and by default has no level
+           handler
+        2. Adding a handler to the root logger affects the custom logger
+        3. When propagation is set to False, only the handlers of the custom
+        logger are checked"""
+        logger = logging.getLogger(module.__name__)
+        assert not logging_handler_defined(logger)
+
+        handler = logging.StreamHandler()
+        logging.root.addHandler(handler)
+        assert logging_handler_defined(logger)
+
+        logger.propagate = False
+        assert not logging_handler_defined(logger)
+
+    def test_create_logger(self, module):
+        """Tests that the create_logger function returns a logger with the
+        default configuration set for an MM logger"""
+        logger = create_logger(module.__name__)
+        assert logger.level == logging.INFO
+        assert logging_handler_defined(logger)
+        assert logger.handlers[0] == default_handler
+
+
+class TestLoggerIntegration:
+    """Tests that the MM logger integrates well with user defined logging
+    configurations."""
+
+    def test_custom_configuration_without_mm_logger(self, tmpdir):
+        """Tests that if there was no MM logger created, custom logging
+        configurations work as expected and no configuration details were set
+        incorrectly."""
+
+        level = logging.DEBUG
+
+        test_file = tmpdir.join("test_file")
+        logging.basicConfig(filename=test_file, level=level)
+        logging.debug("A log entry.")
+
+        assert "A log entry." in test_file.read()
+
+    @pytest.mark.parametrize("module", modules_contain_logging)
+    def test_default_mm_logger(self, module):
+        """Tests that stderr is set for the MM logger by default as stream if
+        there were not other configurations made."""
+
+        logger = create_logger(module.__name__)
+        stream_name = re.search(r"[0-9]", logger.handlers[0].stream.name).group()
+
+        assert len(logger.handlers) == 1
+        # checks if stream is stderr (stream name for stderr is 8 or 9, whereas stdout 6 or 7)
+        assert stream_name in ["8", "9"]
```

Comment:
Seems like the same can be achieved by using the `fileno()` method, instead of applying a regex to find the stream name. Couldn't find any details regarding that method, though, so I'm not exactly sure exactly what it's supposed to do.

```suggestion
        assert len(logger.handlers) == 1
        # checks if stream is stderr (stream name for stderr is 8 or 9, whereas stdout 6 or 7)
        assert logger.handlers[0].stream.fileno() in [8, 9]
```

Reply:
Thanks @thisac, this indeed seem to be a cleaner approach. I couldn't find any documentation or information about the method either, the values here set were found by trial and error on different machines and running using `make test` and `pytest`.